exonum/src/runtime/rust.rs

```diff
@@ -98,44 +113,40 @@ impl RuntimeEnvironment for RustRuntime {
         Ok(())
     }
 
-    fn finish_init(&self, _: &mut EnvContext, _: ServiceInstanceId, abort: bool) {
-        if abort {
-            unimplemented!()
-        }
-    }
-
-    fn execute(&self, context: &mut EnvContext, dispatch: DispatchInfo, payload: &[u8]) {
+    fn execute(
+        &self,
+        context: &mut EnvContext,
+        dispatch: CallInfo,
+        payload: &[u8],
+    ) -> Result<(), ExecutionError> {
         let inner = self.inner.read().unwrap();
         let instance = inner.initialized.get(&dispatch.instance_id).unwrap();
-        let interface = instance.interfaces.get(&dispatch.interface_id).unwrap();
-        let handler = interface.methods.get(dispatch.method_id as usize).unwrap();
+        let handler = instance.methods.get(&dispatch.method_id).unwrap();
 
         let mut ctx = TransactionContext::from_env_ctx(context);
 
-        (handler.fun_untyped)(&mut ctx, payload);
+        (handler.fun_untyped)(&mut ctx, payload)
     }
 }
 
 struct TransactionContext<'a> {
-    env_context: &'a EnvContext<'a>,
+    _env_context: &'a EnvContext<'a>,
 }
 
 impl<'a> TransactionContext<'a> {
     fn from_env_ctx(env_context: &'a EnvContext<'a>) -> Self {
-        Self { env_context }
+        Self {
+            _env_context: env_context,
+        }
     }
 }
 
 struct Handler {
     pub fun_untyped: Box<dyn Fn(&mut TransactionContext, &[u8]) -> Result<(), ExecutionError>>,
```

Comment:
Method name sounds a little bit weird

Reply:
Yea, agree. I guess in the future there will be `fun_typed`, but not sure.
What will you suggest?

Reply:
What about `fn_untyped` or `func_untyped` ? But `fun_...` it's really weird 🙃

Reply:
Sure!